### PR TITLE
fix bug in perception screen

### PR DESCRIPTION
**Pull request by MohmadAyman**
### Description

1. Fixing a bug that makes the SA crashes when switching from the perception screen.
2. Improve documentation.

### Checklist
- [x] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_setup_assistant/src/widgets/perception_widget.cpp [hunk 1]
```diff
@@ -56,24 +56,18 @@ PerceptionWidget::PerceptionWidget(QWidget* parent, moveit_setup_assistant::Move
 
   // Top Header Area ------------------------------------------------
 
-  HeaderWidget* header =
-      new HeaderWidget("3D Perception Sensor Configuration", "Configure your 3D sensors to work with Moveit! ", this);
+  HeaderWidget* header = new HeaderWidget("3D Perception Sensor Configuration",
+                                          "Configure your 3D sensors to work with Moveit! "
+                                          "Look at <a "
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/widgets/perception_widget.cpp]: "Look at" --> "Please see"

File: moveit_setup_assistant/src/widgets/perception_widget.cpp [hunk 2]
```diff
@@ -56,24 +56,18 @@ PerceptionWidget::PerceptionWidget(QWidget* parent, moveit_setup_assistant::Move
...
+                                          "href='http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/"
+                                          "perception_configuration/"
+                                          "perception_configuration_tutorial.html'>Perception Documentation</a> "
+                                          "for more details",
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/widgets/perception_widget.cpp]: Add "." after "details"

**Comment by davetcoleman** [comment]
I still do not see any default values when I choose a sensor plugin:

image

**Comment by davetcoleman** [comment]
Bug: if I go to the "3D Perception" screen and choose "Depth Map" from the drop down list, then switch to the "Passive Joints" screen, the MSA segfaults.

**Comment by MohmadAyman** [comment]
@davetcoleman
I do confirm the other bug about the `segfaults` but I can still not reproduce the no default values. 
Would you tell me what do you see if you run the `catkin test` in the setup assistant? there is actually a test case on reading the default values.

**Comment by MohmadAyman** [comment]
I also cant reproduce the segmentation fault either on this branch.

**Comment by davetcoleman** [comment]
Run on a new moveit_config package, not an existing one.

**Comment by MohmadAyman** [comment]
Fixed the bug the occurs when using on a new package by calling the `input3DSensors` even if the package is new.

**Comment by MohmadAyman** [comment]
@davetcoleman Travis is happy.

File: moveit_setup_assistant/src/tools/moveit_config_data.cpp [hunk 3]
```diff
@@ -1518,6 +1518,7 @@ bool MoveItConfigData::input3DSensorsYAML(const std::string& default_file_path,
       {
```
[the remark this answers is not in the document]
Comment (MohmadAyman) [reply moveit_setup_assistant/src/tools/moveit_config_data.cpp]: I think can do sth like this
```
    // Get sensors node
    const YAML::Node& sensors_node = doc["sensors"];
  
    // Make sure that the sensors are written as a sequence
    if (sensors_node && sensors_node.IsSequence())
```

**Comment by davetcoleman** [comment]
cherry-picked to melodic

**Pull request merged**